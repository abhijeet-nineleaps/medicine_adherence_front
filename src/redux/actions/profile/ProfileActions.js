import { logger } from "react-native-logs";
import Logger from "../../../components/logger";
import Types from "../allTypes";

function saveProfile(user_id) {
  return {
    type: Types.SAVE_PROFILE,
    payload: user_id,
  };
}
function saveProfileSuccess(data) {
  Logger.loggerInfo(data);
  return {
    type: Types.SUCCESS_PROFILE,
    payload: data,
  };
}
function saveProfileFailed(error) {
  Logger.loggerError(error);
  return {
    type: Types.FAILED_PROFILE,
    payload: error,
  };
}

export const ProfileActions = {
  saveProfile,
  saveProfileSuccess,
  saveProfileFailed,
}
