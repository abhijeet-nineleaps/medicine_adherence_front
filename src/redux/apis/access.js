import {API_URLS} from '../../constants/apiUrl';
export const Signupuser = {
  signup: async (params) => {
    const {userinfo, token} = params;
    let url = new URL(API_URLS.SIGNUP);
    url.searchParams.append('fcmToken', token);
    url.searchParams.append('picPath', userinfo.user.photo);

    const response = await fetch(url, {
      method: 'POST',
      body: JSON.stringify({
        userName: userinfo.user.givenName,
        email: userinfo.user.email,
      }),
      headers: {
        'Content-type': 'application/json',
      },
    });
    return response;
  },
  loginuser: async (params) => {
    const {userinfo, token} = params;
    let url = new URL(API_URLS.LOGIN);
    const response = await fetch(url, {
      method: 'POST',
      body: JSON.stringify({
        email: userinfo.user.email,
        fcmToken: token,
      }),
      headers: {
        'Content-type': 'application/json',
      },
    });
    return response;
  },
};
