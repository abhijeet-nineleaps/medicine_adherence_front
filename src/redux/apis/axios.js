import defaultAxios from 'axios';
import { API_URL } from '../../repositories/var';

const axios = defaultAxios.create({
  baseURL: `${API_URL}`,
  headers: {'Content-Type': 'application/json'},
});

export default axios;
