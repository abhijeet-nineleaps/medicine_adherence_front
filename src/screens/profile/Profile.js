import {GoogleSignin} from '@react-native-google-signin/google-signin';
import React from 'react';
import {Text, View, Image, ScrollView, Alert} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {Picker} from '@react-native-picker/picker';
import {TextInput} from 'react-native-paper';
import Toast from 'react-native-toast-message';
import {API_URL} from '../../repositories/var';
import {Button} from 'react-native-elements';
import {Formik} from 'formik';
import * as yup from 'yup';
import styles from './profileStyles/ProfileStyles';
import SavedDetails from './SavedDetails';

import {useFocusEffect} from '@react-navigation/native';

import Icon from 'react-native-vector-icons/FontAwesome5';
import OctIcon from 'react-native-vector-icons/Octicons';
import IonIcon from 'react-native-vector-icons/Ionicons';
import EntIcon from 'react-native-vector-icons/Entypo';

const loginValidationSchema = yup.object().shape({
  Bio: yup.string().required('Bio is Required'),
  Contact: yup
    .string()
    .min(10, ({min}) => `Contact number must be ${min} characters`)
    .max(10, ({max}) => `Contact number can be only ${max} characters`)
    .required('Contact is Required'),
  Age: yup.string().min(2, ({min}) => `Age number must be ${min} characters`),
  Weight: yup
    .string()
    .min(2, ({min}) => `Weight must be at least ${min} characters`),
  Gender: yup
    .string()
    .min(3, ({min}) => `Gender must be at least ${min} characters`)
    .required(),
  MaritalStatus: yup
    .string()
    .min(3, ({min}) => `MaritalStatus must be at least ${min} characters`)
    .required(),
  BloodGroup: yup.string().required(),
});

const Profile = ({navigation}) => {
  const [name, namestate] = React.useState({
    user: {name: 'Not logged in!', photo: '', email: ''},
  });
  const [img, imgstate] = React.useState('https://i.stack.imgur.com/l60Hf.png');

  const [load, loadstate] = React.useState(false);

  const [editenabled, editstate] = React.useState(false);

  async function storeuserdetails(values) {
    loadstate(true);
    await AsyncStorage.setItem('bio', values.Bio);
    await AsyncStorage.setItem('contact', values.Contact);
    await AsyncStorage.setItem('age', values.Age);
    await AsyncStorage.setItem('weight', values.Weight);
    await AsyncStorage.setItem('gender', values.Gender);
    await AsyncStorage.setItem('maritalstatus', values.MaritalStatus);
    await AsyncStorage.setItem('bloodgroup', values.BloodGroup);

    let sbio = await AsyncStorage.getItem('bio');
    let scontact = await AsyncStorage.getItem('contact');
    let sage = await AsyncStorage.getItem('age');
    let sweight = await AsyncStorage.getItem('weight');
    let sgender = await AsyncStorage.getItem('gender');
    let maritalstatus = await AsyncStorage.getItem('maritalstatus');
    let sblood = await AsyncStorage.getItem('bloodgroup');

    const user_id = await AsyncStorage.getItem('user_id');

    await fetch(`${API_URL}/api/v1/user-details?userId=${user_id}`, {
      method: 'PUT',
      body: JSON.stringify({
        bio: sbio,
        age: sage,
        userContact: scontact,
        gender: sgender,
        martialStatus: maritalstatus,
        bloodGroup: sblood,
        weight: sweight,
      }),
      headers: {
        'Content-type': 'application/json',
      },
    })
      .then(resp => {
        if (resp.status === 200) {
          loadstate(false);
          editstate(false);
        }
      })
      .then(() => {
        Toast.show({
          type: 'success',
          text1: 'Details saved',
        });
      });

    Toast.show({
      type: 'info',
      text1: 'Please fill details properly',
    });
  }
  useFocusEffect(
    React.useCallback(() => {
      async function getuser() {
        try {
          if (!(await GoogleSignin.isSignedIn())) {
            Alert.alert(
              'Sign in first to Edit Profile',
              'Click ok to proceed',
              [
                {
                  text: 'Ok',
                  onPress: () => {
                    navigation.navigate('Login');
                  },
                },
                {
                  text: 'Cancel',
                  onPress: () => {
                    navigation.navigate('Home');
                  },
                },
              ],
            );
            return;
          }
          const user = await GoogleSignin.getCurrentUser();

          namestate(user);
          imgstate(user.user.photo);
        } catch (err) {}
      }
      getuser();
      return () => {
        /* do nothing */
      };
    }, []),
  );
  const weightIcon = () => {
    return <Icon size={18} name="weight" color="#3743ab"></Icon>;
  };
  const userIcon = () => {
    return <Icon size={18} name="user" color="#3743ab"></Icon>;
  };
  const phoneIcon = () => {
    return <Icon size={18} name="phone-alt" color="#3743ab"></Icon>;
  };
  return (
    <View style={styles.container}>
      <ScrollView>
        <Toast visibilityTime={1500}></Toast>
        <View style={styles.container1}>
          <View style={styles.top}>
            <Image source={{uri: img}} style={styles.avatar}></Image>
            <View style={styles.topItem}>
              <Text style={styles.topItemText1}>{name.user.name}</Text>
              <Text style={styles.topItemText2}>{name.user.email}</Text>
            </View>
          </View>
          <View>
            <Button
              id="edit"
              title="Edit profile"
              buttonStyle={styles.editButton}
              containerStyle={styles.editButtonConatiner}
              onPress={() => editstate(true)}></Button>
          </View>
          <View>
            <View>
              {editenabled ? (
                <>
                  <View>
                    <Formik
                      validationSchema={loginValidationSchema}
                      initialValues={{
                        Bio: '',
                        Contact: '',
                        Age: '',
                        Weight: '',
                        Gender: '',
                        MaritalStatus: '',
                        BloodGroup: '',
                      }}
                      onSubmit={values => storeuserdetails(values)}>
                      {({
                        handleChange,
                        handleBlur,
                        handleSubmit,
                        values,
                        errors,
                        isValid,
                        touched,
                        setFieldValue,
                      }) => (
                        <View style={styles.inputContainer}>
                          <TextInput
                            label=" Bio"
                            mode="flat"
                            style={styles.textInput}
                            id=""
                            onChangeText={handleChange('Bio')}
                            onBlur={handleBlur('Bio')}
                            value={values.Bio}
                            left={<TextInput.Icon name={userIcon} />}
                          />
                          {errors.Bio && touched.Bio && (
                            <Text style={styles.errorText}>{errors.Bio}</Text>
                          )}
                          <TextInput
                            label=" Contact"
                            mode="flat"
                            keyboardType="numeric"
                            style={styles.textInput}
                            onChangeText={handleChange('Contact')}
                            onBlur={handleBlur('Contact')}
                            value={values.Contact}
                            left={<TextInput.Icon name={phoneIcon} />}
                          />
                          {errors.Contact && touched.Contact && (
                            <Text style={styles.errorText}>
                              {errors.Contact}
                            </Text>
                          )}
                          <TextInput
                            label=" Age(in years)"
                            mode="flat"
                            keyboardType="numeric"
                            style={styles.textInput}
                            onChangeText={handleChange('Age')}
                            onBlur={handleBlur('Age')}
                            value={values.Age}
                            left={
                              <TextInput.Icon
                                name={() => (
                                  <OctIcon
                                    size={18}
                                    name="number"
                                    color="#3743ab"></OctIcon>
                                )}
                              />
                            }
                          />
                          {errors.Age && (
                            <Text style={styles.errorText}>{errors.Age}</Text>
                          )}
                          <TextInput
                            label=" Weight(in kg)"
                            mode="flat"
                            keyboardType="numeric"
                            style={styles.textInput}
                            onChangeText={handleChange('Weight')}
                            onBlur={handleBlur('Weight')}
                            value={values.Weight}
                            left={<TextInput.Icon name={weightIcon} />}
                          />
                          {errors.Weight && (
                            <Text style={styles.errorText}>
                              {errors.Weight}
                            </Text>
                          )}
                          <View style={styles.pickerContainer}>
                            <View style={styles.pickerIcon}>
                              <IonIcon
                                size={18}
                                name="male-female"
                                color="#3743ab"></IonIcon>
                            </View>
                            <View style={styles.picker}>
                              <Picker
                                mode="dropdown"
                                selectedValue={values.Gender}
                                onValueChange={itemchange =>
                                  setFieldValue('Gender', itemchange)
                                }>
                                <Picker.Item
                                  label="Gender"
                                  value="Gender"
                                  style={styles.pickerItem}
                                />
                                <Picker.Item label="Male" value="Male" />
                                <Picker.Item label="Female" value="Female" />
                                <Picker.Item label="Other" value="Other" />
                              </Picker>
                            </View>
                            <Text style={styles.pickerText}>
                              {touched.Gender && errors.Gender}
                            </Text>
                          </View>
                          <View style={styles.pickerContainer}>
                            <View style={styles.pickerIcon}>
                              <Icon
                                size={18}
                                name="ring"
                                color="#3743ab"></Icon>
                            </View>
                            <View style={styles.picker}>
                              <Picker
                                mode="dropdown"
                                selectedValue={values.MaritalStatus}
                                onValueChange={itemchange =>
                                  setFieldValue('MaritalStatus', itemchange)
                                }>
                                <Picker.Item
                                  label="Marital Status"
                                  value="Marital Status"
                                  style={styles.pickerItem}
                                />

                                <Picker.Item label="Married" value="Married" />
                                <Picker.Item
                                  label="Unmarried"
                                  value="Unmarried"
                                />
                              </Picker>
                            </View>
                            <Text style={styles.pickerText}>
                              {touched.MaritalStatus && errors.MaritalStatus}
                            </Text>
                          </View>
                          <View style={styles.pickerContainer}>
                            <View style={styles.pickerIcon}>
                              <EntIcon
                                size={18}
                                name="drop"
                                color="#3743ab"></EntIcon>
                            </View>
                            <View style={styles.bgPickerView}>
                              <Picker
                                mode="dropdown"
                                style={styles.bgPicker}
                                selectedValue={values.BloodGroup}
                                onValueChange={itemchange =>
                                  setFieldValue('BloodGroup', itemchange)
                                }>
                                <Picker.Item
                                  label="BloodGroup"
                                  value="BloodGroup"
                                  style={styles.pickerItem}
                                />

                                <Picker.Item label="A+" value="A+" />
                                <Picker.Item label="A-" value="A" />
                                <Picker.Item label="B+" value="B+" />
                                <Picker.Item label="B-" value="B-" />
                                <Picker.Item label="O+" value="O+" />
                                <Picker.Item label="O-" value="O-" />
                                <Picker.Item label="AB+" value="AB+" />
                                <Picker.Item label="AB-" value="AB-" />
                              </Picker>
                            </View>
                            <Text style={styles.pickerText}>
                              {touched.BloodGroup && errors.BloodGroup}
                            </Text>
                          </View>
                          <View style={styles.saveButtonView}>
                            <Button
                              id="save"
                              loading={load}
                              onPress={handleSubmit}
                              title="SAVE"
                              buttonStyle={styles.saveButton}
                              containerStyle={styles.saveButtonContainer}
                              disabled={!isValid}
                            />
                          </View>
                        </View>
                      )}
                    </Formik>
                  </View>
                </>
              ) : (
                <SavedDetails />
              )}
            </View>
          </View>
        </View>
      </ScrollView>
    </View>
  );
};

export default Profile;
