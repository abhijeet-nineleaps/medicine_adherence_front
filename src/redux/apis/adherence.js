import axios from './axios';
import { API_URLS } from '../../constants/apiUrl';

const adherence = {
  syncmedicineHistory: async (params) => {
    const {meds_id, syncData} = params;
    let response = await fetch(`${API_URLS.SYNC_HISTORY}?medId=${meds_id}`, {
      method: 'POST',
      body: JSON.stringify(syncData),
      headers: {
        'Content-type': 'application/json',
      },
    });
    return response;
  },
  medimages: async function fetchmedimages(payload) {
    const response = await axios.get(`${API_URLS.MED_IMAGES}?medId=${payload}`);
    return response.data;
  },
  downloadPdf: async function downloadPdf(payload) {
    const response = await axios.get(`${API_URLS.DOWNLOAD_PDF}?medId=${payload}`);
    return response.data;
  },
  syncmeds: async function syncMeds(payload) {
    const response = await axios.get(`${API_URLS.SYNC_MEDS}=${payload}`);
    return response.data;
  },
  getmedhistory: async function getMedHistory(payload) {
    const response = await axios.get(`${API_URLS.GET_MED_HISTORY}?medId=${payload}`);
    return response.data;
  }

};


export default adherence;
