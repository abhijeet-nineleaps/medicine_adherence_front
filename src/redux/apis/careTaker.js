import axios from './axios';
import { API_URLS } from '../../constants/apiUrl';
export const careTaker = {
  emailcaretaker: async function name(payload) {
    const response = await axios.get(
      `${API_URLS.EMAIL_CARETAKER}?email=${payload}&sender={udet.user.givenName}`,
    );
    return response.data;
  },
  caretaker: async function fetchcaretaker(payload) {
    const response = await axios.get(
      `${API_URLS.CARETAKER}?patientId=${payload}`,
    );
    return response.data;
  },
  reqCaretaker: async function reqcaretaker(payload) {
    const response = await axios.get(`${API_URLS.CARETAER_REQUEST}`);
    return response.data;
  },
  sendImage: async function sendimages(payload) {
    const response = await axios.post(
      `${API_URLS.SEND_IMAGE}?medId=${payload}`,
    );
    return response.data;
  },
};
