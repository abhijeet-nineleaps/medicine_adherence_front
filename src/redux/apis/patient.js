import axios from './axios';
import { API_URLS } from '../../constants/apiUrl';
export const patient = {
 fetchPatient: async function fetchPatient(payload) {
    const response = await axios.get(`${API_URLS.GET_PATIENT}?caretakerId=${payload}`);
    return response.data;
  },
  patientProfile: async function fetchpatientdetail(payload) {
    const response = await axios.get(`${API_URLS.PATIENT_PROFILE}?userId=${payload}`);
    return response.data;
  },
  patientReq: async function fetchpatientreq(payload) {
    const response = await axios.get(`${API_URLS.PATIENT_REQ}?userId=${payload}`);
    return response.data;
  },
  reqAccept: async function fetchpatientreqaccept(payload) {
    const response = await axios.get(`${API_URLS.REQ_ACCEPT}`);
    return response.data;
  },
  reqDelete: async function fetchpatientreqdelete(payload) {
    const response = await axios.get(`${API_URLS.REQ_DELETE}`);
    return response.data;
  },
  notifyPatient: async function notifypatient(payload) {
    const response = await axios.put(`${API_URLS.NOTIFY_PATIENT}?medname=${payload}`);
    return response.data;
  },
}