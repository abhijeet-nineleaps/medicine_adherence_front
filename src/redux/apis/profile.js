import axios from './axios';
import { API_URLS } from '../../constants/apiUrl';
export const profile = {
    saveProfile: async function saveprofile(payload) {
        const response = await axios.put(`${API_URLS.SAVE_PROFILE}?userId=${payload}`);
        return response.data;
      }
}