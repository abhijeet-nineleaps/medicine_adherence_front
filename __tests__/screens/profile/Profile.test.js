
import React from 'react';
import renderer from 'react-test-renderer';
import { Formik } from 'formik';
import {Picker} from '@react-native-picker/picker';
import toJson from 'enzyme-to-json';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Profile from '../../../src/screens/profile/Profile';
import { profile } from '../../../src/redux/apis/profile';
import { TextInput } from 'react-native-paper';
import { render } from '@testing-library/react-native';
import Mypatient from '../../../src/screens/patient/MyPatients';
Enzyme.configure({adapter: new Adapter()});
jest.mock('@react-native-google-signin/google-signin', () => {
  return {
  GoogleSignin: jest.fn().mockImplementation(() => {
  return null;
  }),
  };
  });
global.alert=jest.fn();
jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useFocusEffect: jest.fn().mockImplementation((func) => func()),
  
}));
const props = {
  navigation: {
    addListener: jest.fn(),
    navigate: jest.fn(),
    Bio: jest.fn(),
    Contact: jest.fn(),
    Age: jest.fn(),
    Weight: jest.fn(),
    Gender: jest.fn(),
    MaritalStatus: jest.fn(),
    BloodGroup: jest.fn(),
  },
};
const findNodeByTestId = (wrapper, testID) => {
  return wrapper.findWhere((node) => {
    return node.prop("testID") === testID;
  });
}
describe('Click send image', () => {
  it('renders correctly', () => {
    // const tree = renderer
    //   .create(<Profile navigation={undefined}/>)
    //   .childAt(0).dive();
     
    // expect(tree).toMatchSnapshot();


    const wrapper = shallow(<Profile/>);
    expect.assertions(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('test open edit button', () => {
    const mockFn = jest.fn();
   const wrapper = shallow(<Profile editstate={mockFn} />);
   wrapper.find("#edit").simulate("press");
   });
   
});
describe('<FirstComponent /> functions', () => {
  it('test the only function', () => {
    
      const wrapper = renderer.create(<Profile/>);
      const inst = wrapper.getInstance();
      expect(inst?.userIcon(<Icon name="home" size={22} color="white"/>)).toMatchSnapshot();
  });
  it('test the only function', () => {
  
    const wrapper = renderer.create(<Profile/>);
    const inst = wrapper.getInstance();
    expect(inst?.weightIcon(<Icon size={18} name="weight" color="#3743ab"></Icon>)).toMatchSnapshot();
});
it('test the only function', () => { 
  const wrapper = renderer.create(<Profile/>);
  const inst = wrapper.getInstance();
  expect(inst?.phoneIcon(<Icon name="home" size={22} color="white"/>)).toMatchSnapshot();
});
it("test for editstate", () => {
  const wrapper = shallow(<Profile navigation={props.navigation} />);
  const tree = findNodeByTestId(wrapper, "estate");
  tree.props().onPress();
});
// it("test for userIcon", () => {
//   const editenabled = true;

//     React.useState = jest.fn().mockReturnValue([editenabled, {}]);
//   const wrapper = shallow(<Profile navigation={props.navigation} />);
//   const tree = findNodeByTestId(wrapper, "userID");
//   tree.props().left();
// });
// it("test for phoneIcon", () => {
  
//   const wrapper = shallow(<Profile navigation={props.navigation} />);
//   const tree = findNodeByTestId(wrapper, "phoneID");
//   tree.props().left();
// });
// it("test for weightIcon", () => {
//   const editenabled = true;
//   const name={user: {name: 'Not logged in!', photo: '', email: ''}}

//     React.useState = jest.fn().mockReturnValue([editenabled, {}]);
//     React.useState = jest.fn().mockReturnValue([name, {}]);
  
//     const wrapper = shallow(<Profile navigation={props.navigation} />);
//   const tree = findNodeByTestId(wrapper, "weightID");
//   console.log(wrapper.debug(),"30")
//   // tree.props().left();
// });
jest.mock("@react-native-picker/picker", () => props => (
  <input
  testID="marital"
    onValueChange={() => {
      console.log(props.onChange);
      props.onValueChange("maritalstatus");
    }}
  />
));
const storeuserdetails = jest.fn();
  
  it("button click ", () => {
    const component = shallow(<Profile onSubmit={storeuserdetails} />);
     component.find("Formik").simulate("submit");
    expect(component).toMatchSnapshot();
    component.unmount();
  });
//   describe('Click send image', () => {
//   it("should call onDeleteList", () => {
//     const wrapper = shallow(<Profile navigation={props.navigation} />);
//     const tree = findNodeByTestId(wrapper, "log");
//     tree.props().validationSchema;
//   });
// });
});



