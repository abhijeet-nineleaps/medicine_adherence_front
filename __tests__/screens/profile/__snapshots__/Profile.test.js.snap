// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstComponent /> functions button click  1`] = `ShallowWrapper {}`;

exports[`<FirstComponent /> functions test the only function 1`] = `undefined`;

exports[`<FirstComponent /> functions test the only function 2`] = `undefined`;

exports[`<FirstComponent /> functions test the only function 3`] = `undefined`;

exports[`Click send image renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": "100%",
    }
  }
>
  <ScrollView>
    <Toast
      visibilityTime={1500}
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flexDirection": "column",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3743ab",
            "borderBottomLeftRadius": 30,
            "borderBottomRightRadius": 30,
            "justifyContent": "center",
            "marginBottom": 12,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://i.stack.imgur.com/l60Hf.png",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "borderColor": "white",
              "borderRadius": 83,
              "borderWidth": 4,
              "height": 100,
              "marginBottom": 10,
              "marginTop": 30,
              "width": 100,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "marginBottom": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            Not logged in!
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          />
        </View>
      </View>
      <View>
        <Themed.Button
          buttonStyle={
            Object {
              "backgroundColor": "#3743ab",
              "borderRadius": 30,
              "width": "48%",
            }
          }
          containerStyle={
            Object {
              "alignItems": "center",
            }
          }
          id="edit"
          onPress={[Function]}
          testID="estate"
          title="Edit profile"
        />
      </View>
      <View>
        <View>
          <View>
            <Formik
              initialValues={
                Object {
                  "Age": "",
                  "Bio": "",
                  "BloodGroup": "",
                  "Contact": "",
                  "Gender": "",
                  "MaritalStatus": "",
                  "Weight": "",
                }
              }
              onSubmit={[Function]}
              testID="log"
              validationSchema={
                ObjectSchema {
                  "_blacklist": ReferenceSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_blacklistError": undefined,
                  "_excludedEdges": Array [],
                  "_nodes": Array [
                    "BloodGroup",
                    "MaritalStatus",
                    "Gender",
                    "Weight",
                    "Age",
                    "Contact",
                    "Bio",
                  ],
                  "_sortErrors": [Function],
                  "_typeError": [Function],
                  "_whitelist": ReferenceSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_whitelistError": undefined,
                  "conditions": Array [],
                  "deps": Array [],
                  "exclusiveTests": Object {},
                  "fields": Object {
                    "Age": StringSchema {
                      "_blacklist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_blacklistError": undefined,
                      "_typeError": [Function],
                      "_whitelist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_whitelistError": undefined,
                      "conditions": Array [],
                      "deps": Array [],
                      "exclusiveTests": Object {
                        "min": true,
                      },
                      "spec": Object {
                        "abortEarly": true,
                        "nullable": false,
                        "presence": "optional",
                        "recursive": true,
                        "strict": false,
                        "strip": false,
                      },
                      "tests": Array [
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                      "type": "string",
                    },
                    "Bio": StringSchema {
                      "_blacklist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_blacklistError": undefined,
                      "_mutate": undefined,
                      "_typeError": [Function],
                      "_whitelist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_whitelistError": undefined,
                      "conditions": Array [],
                      "deps": Array [],
                      "exclusiveTests": Object {
                        "required": true,
                      },
                      "spec": Object {
                        "abortEarly": true,
                        "nullable": false,
                        "presence": "required",
                        "recursive": true,
                        "strict": false,
                        "strip": false,
                      },
                      "tests": Array [
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                      "type": "string",
                    },
                    "BloodGroup": StringSchema {
                      "_blacklist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_blacklistError": undefined,
                      "_mutate": undefined,
                      "_typeError": [Function],
                      "_whitelist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_whitelistError": undefined,
                      "conditions": Array [],
                      "deps": Array [],
                      "exclusiveTests": Object {
                        "required": true,
                      },
                      "spec": Object {
                        "abortEarly": true,
                        "nullable": false,
                        "presence": "required",
                        "recursive": true,
                        "strict": false,
                        "strip": false,
                      },
                      "tests": Array [
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                      "type": "string",
                    },
                    "Contact": StringSchema {
                      "_blacklist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_blacklistError": undefined,
                      "_mutate": undefined,
                      "_typeError": [Function],
                      "_whitelist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_whitelistError": undefined,
                      "conditions": Array [],
                      "deps": Array [],
                      "exclusiveTests": Object {
                        "max": true,
                        "min": true,
                        "required": true,
                      },
                      "spec": Object {
                        "abortEarly": true,
                        "nullable": false,
                        "presence": "required",
                        "recursive": true,
                        "strict": false,
                        "strip": false,
                      },
                      "tests": Array [
                        [Function],
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                      "type": "string",
                    },
                    "Gender": StringSchema {
                      "_blacklist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_blacklistError": undefined,
                      "_mutate": undefined,
                      "_typeError": [Function],
                      "_whitelist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_whitelistError": undefined,
                      "conditions": Array [],
                      "deps": Array [],
                      "exclusiveTests": Object {
                        "min": true,
                        "required": true,
                      },
                      "spec": Object {
                        "abortEarly": true,
                        "nullable": false,
                        "presence": "required",
                        "recursive": true,
                        "strict": false,
                        "strip": false,
                      },
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                      "type": "string",
                    },
                    "MaritalStatus": StringSchema {
                      "_blacklist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_blacklistError": undefined,
                      "_mutate": undefined,
                      "_typeError": [Function],
                      "_whitelist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_whitelistError": undefined,
                      "conditions": Array [],
                      "deps": Array [],
                      "exclusiveTests": Object {
                        "min": true,
                        "required": true,
                      },
                      "spec": Object {
                        "abortEarly": true,
                        "nullable": false,
                        "presence": "required",
                        "recursive": true,
                        "strict": false,
                        "strip": false,
                      },
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                      "type": "string",
                    },
                    "Weight": StringSchema {
                      "_blacklist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_blacklistError": undefined,
                      "_typeError": [Function],
                      "_whitelist": ReferenceSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_whitelistError": undefined,
                      "conditions": Array [],
                      "deps": Array [],
                      "exclusiveTests": Object {
                        "min": true,
                      },
                      "spec": Object {
                        "abortEarly": true,
                        "nullable": false,
                        "presence": "optional",
                        "recursive": true,
                        "strict": false,
                        "strip": false,
                      },
                      "tests": Array [
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                      "type": "string",
                    },
                  },
                  "spec": Object {
                    "abortEarly": true,
                    "nullable": false,
                    "presence": "optional",
                    "recursive": true,
                    "strict": false,
                    "strip": false,
                  },
                  "tests": Array [],
                  "transforms": Array [
                    [Function],
                  ],
                  "type": "object",
                }
              }
            >
              <Component />
            </Formik>
          </View>
        </View>
      </View>
    </View>
  </ScrollView>
</View>
`;
